#Self

Name - Soumya Bairagya

College - VIT, Pune

E-mail - bairagyasoumya2001@gmail.com


#**Assignment 2 - Multivariate Linear Regression**
###Employee Hiring Salary Prediction

In [ ]:
#Importing the required libraries
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression
#installing the word2number library and then importing w2n
!pip install word2number
from word2number import w2n

In [ ]:
#Reading and printing the csv file
hdf = pd.read_csv("hiring.csv")
hdf

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0        NaN                    8.0                           9      50000
1        NaN                    8.0                           6      45000
2       five                    6.0                           7      60000
3        two                   10.0                          10      65000
4      seven                    9.0                           6      70000
5      three                    7.0                          10      62000
6        ten                    NaN                           7      72000
7     eleven                    7.0                           8      80000

In [ ]:
#Knowing the columns in the dataframe
hdf.columns

Index(['experience', 'test_score(out of 10)', 'interview_score(out of 10)',
       'salary($)'],
      dtype='object')

In [ ]:
#Filling the na values in experience column with zero
hdf.experience = hdf.experience.fillna("zero")
hdf

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0       zero                    8.0                           9      50000
1       zero                    8.0                           6      45000
2       five                    6.0                           7      60000
3        two                   10.0                          10      65000
4      seven                    9.0                           6      70000
5      three                    7.0                          10      62000
6        ten                    NaN                           7      72000
7     eleven                    7.0                           8      80000

In [ ]:
#Using the w2n.wor_to_num function to convert the words(number sentence) into number
hdf.experience = hdf.experience.apply(w2n.word_to_num)
hdf

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0           0                    8.0                           9      50000
1           0                    8.0                           6      45000
2           5                    6.0                           7      60000
3           2                   10.0                          10      65000
4           7                    9.0                           6      70000
5           3                    7.0                          10      62000
6          10                    NaN                           7      72000
7          11                    7.0                           8      80000

In [ ]:
#Finding the median value of the test_score(out of 10) column
m = hdf['test_score(out of 10)'].median()
m

8.0

In [ ]:
#Filling the na value in test_score(out of 10) column with its median value
hdf['test_score(out of 10)'] = hdf['test_score(out of 10)'].fillna(m)

##Final Dataframe

In [ ]:
hdf

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0           0                    8.0                           9      50000
1           0                    8.0                           6      45000
2           5                    6.0                           7      60000
3           2                   10.0                          10      65000
4           7                    9.0                           6      70000
5           3                    7.0                          10      62000
6          10                    8.0                           7      72000
7          11                    7.0                           8      80000

In [ ]:
#Separating the Independent columns
X = hdf.drop(columns="salary($)")
X

experience  test_score(out of 10)  interview_score(out of 10)
0           0                    8.0                           9
1           0                    8.0                           6
2           5                    6.0                           7
3           2                   10.0                          10
4           7                    9.0                           6
5           3                    7.0                          10
6          10                    8.0                           7
7          11                    7.0                           8

In [ ]:
#The dependent value column
Y = hdf["salary($)"]
Y

0    50000
1    45000
2    60000
3    65000
4    70000
5    62000
6    72000
7    80000
Name: salary($), dtype: int64

##Training the data so that model will be ready for prediction

In [ ]:
from sklearn.linear_model import LinearRegression
model = LinearRegression()

In [ ]:
#Training the data so that model will be ready for prediction
M = model.fit(X,Y)

###Coefficient(Slope) and Intercept

In [ ]:
#Slope
M.coef_

array([2812.95487627, 1845.70596798, 2205.24017467])

In [ ]:
#Intercept
M.intercept_

17737.263464337695

###Accuracy of model

In [ ]:
#Accuracy of the model in percentage
M.score(X,Y) * 100

96.1678000257796

###*Prediction and Cross verification*

Formula for multivariate Linear Regression

**y = m1x1 + m2x2 + m3x3 + b**

Example 1

In [ ]:
M.predict([[8,7,10]])

array([75213.24599709])

In [ ]:
#y = m1x1 + m2x2 + m3x3 + b
(2812.95487627*(8)) + (1845.70596798*(7)) + (2205.24017467*(10)) + (17737.263464337695)

75213.2459970577

Example 2

In [ ]:
M.predict([[2,10,10]])

array([63872.63464338])

In [ ]:
#y = m1x1 + m2x2 + m3x3 + b
(2812.95487627*(2)) + (1845.70596798*(10)) + (2205.24017467*(10)) + (17737.263464337695)

63872.63464337769

###Employee Hiring Salary Prediction

This file contains hiring statics for a firm such as experience of candidate, his written test score and personal interview score. Based on these 3 factors, HR will decide the salary. Given this data, you need to build a machine learning model for HR department that can help them decide salaries for future candidates. Using this predict salaries for following candidates
1. 2 yr experience, 9 test score, 6 interview score
2. 12 yr experience, 10 test score, 10 interview score

1. 2 yr experience, 9 test score, 6 interview score

In [ ]:
M.predict([[2,9,6]])

array([53205.96797671])

In [ ]:
#y = m1x1 + m2x2 + m3x3 + b
(2812.95487627*(2)) + (1845.70596798*(9)) + (2205.24017467*(6)) + (17737.263464337695)

53205.9679767177

2. 12 yr experience, 10 test score, 10 interview score

In [ ]:
M.predict([[12,10,10]])

array([92002.18340611])

In [ ]:
#y = m1x1 + m2x2 + m3x3 + b
(2812.95487627*(12)) + (1845.70596798*(10)) + (2205.24017467*(10)) + (17737.263464337695)

92002.18340607769

***Conclusion***

Hence,
1. Salary of the candidate with 2 yr experience, 9 test score, 6 interview score is ***53205.96797671***
2. Salary of the candidate with 12 yr experience, 10 test score, 10 interview score ***92002.18340611***